packages/next-on-pages/templates/_worker.js/index.ts

```diff
@@ -21,10 +21,13 @@ export default {
 	async fetch(request, env, ctx) {
 		const envAsyncLocalStorage = await __ENV_ALS_PROMISE__;
 		if (!envAsyncLocalStorage) {
-			return new Response(
-				`Error: Could not access built-in Node.js modules. Please make sure that your Cloudflare Pages project has the 'nodejs_compat' compatibility flag set.`,
-				{ status: 503 },
+			const noNodeJsCompatStaticPageRequest = await fetch(
+				'https://no-nodejs-compat-flag-static-error-page.pages.dev/',
```

Comment:
I understand why we do this but i also dislike it. I can't think of anything better though so it's cool.

Reply:
yeah.... it's kinda dirty... but I doubt people would be too happy to carry an extra (unnecessary) html + assets in their applications for no good reason 😅

let's hope that this is anyways a temporary patch until we can fail at build time (so hopefully we will remove this runtime check altogether in a future major)